include/exiv2/asfvideo.hpp

```diff
@@ -47,6 +48,41 @@ class EXIV2API AsfVideo : public Image {
   //@{
   [[nodiscard]] std::string mimeType() const override;
   //@}
+
+  /* @class GUID_struct
+   *
+   * @brief A class to represent a globally unique identifier (GUID) structure
+   *
+   * This class represents a globally unique identifier (GUID) structure which is used to identify objects in a
+   * distributed environment. A GUID is a unique identifier that is generated on a computer and can be used to
+   * identify an object across different systems. The GUID structure is comprised of four 32-bit values and an
+   * array of 8 bytes.
+   *
+   * @note The byte order of the GUID structure is in little endian.
+   *
+   * @see https://en.wikipedia.org/wiki/Globally_unique_identifier
+   *
+   */
+  class GUIDTag {
+    uint32_t data1_;
+    uint16_t data2_;
+    uint16_t data3_;
+    std::array<byte, 8> data4_;
+
+   public:
+    bool operator==(const GUIDTag& other) const;
+
+    // Constructor to create a GUID object by passing individual values for each attribute
+    GUIDTag(unsigned int data1, unsigned short data2, unsigned short data3, std::array<byte, 8> data4);
```

Comment:
I don't think this is needed, {} can be used instead of ().

Reply:
but here I already have " GUIDTag(const uint8_t* bytes);" witch will delete the default initializer list

```
exiv2/src/asfvideo.cpp:92:25: error: no matching constructor for initialization of 'const AsfVideo::GUIDTag'
const AsfVideo::GUIDTag Header = {0x75B22630, 0x668E, 0x11CF, {0xA6, 0xD9, 0x00, 0xAA, 0x00, 0x62, 0xCE, 0x6C}};
```

Reply:
that error comes from the std::array.

```
const AsfVideo::GUIDTag Header = {0x75B22630, 0x668E, 0x11CF, std::array<byte, 8>{0xA6, 0xD9, 0x00, 0xAA, 0x00, 0x62, 0xCE, 0x6C}};
```

should work. or don't use std::array. I can handle this in a later PR if you want.